addons/website/models/mixins.py

```diff
@@ -200,11 +200,14 @@ def website_publish_button(self):
         return self.write({'website_published': not self.website_published})
 
     def open_website_url(self):
-        url = self.env['website'].get_client_action_url(self.website_url)
         return {
-            'type': 'ir.actions.act_url',
-            'url': url,
-            'target': 'self',
+            'type': 'ir.actions.client',
+            'tag': 'website_preview',
+            'context': {
+                'params': {
+                    'path': self.website_url,
+                },
+            },
```

Comment:
A while ago, I suggested to @rdeodoo to create a `get_client_action` method on the website modal (next to `get_client_action_url` (maybe in the future to remove get_client_action_url completely). Maybe you could do it in this PR? (the method would receive a path and potential other things like "enable_editor", etc and return this client action object description like here :+1: (you can then adapt the code everywhere you do this).